cmd/run/run.go

```diff
@@ -504,6 +506,14 @@ func (s *ServerContext) Run(ctx context.Context, config *serverconfig.Config) er
 		server.WithRequestDurationByQueryHistogramBuckets(convertStringArrayToUintArray(config.RequestDurationDatastoreQueryCountBuckets)),
 		server.WithMaxAuthorizationModelSizeInBytes(config.MaxAuthorizationModelSizeInBytes),
 		server.WithExperimentals(experimentals...),
+		server.WithTypesystemResolver(
+			typesystem.NewCachedTypesystemResolver(
+				datastore,
+				typesystem.WithTypesystemResolverCache(
+					ccache.New(ccache.Configure[*typesystem.TypeSystem]().MaxSize(int64(config.ModelCacheMaxSizeBytes))),
```

Comment:
This reads very weird. 

```go
typesystem.NewCachedTypesystemResolver(
....typesystem.WithTypesystemResolverCache(
```

Reply:
Constructing a CachedTypesystemResolver and providing a cache to do so.

Would `typesystem.NewCachedTypesystemResolver(typesystem.WithCache(...)` read better?

Reply:
> Constructing a CachedTypesystemResolver and providing a cache to do so.

Then why not simply `NewCachedTypesystemResolver(size)` and

```go
func NewCachedTypesystemResolver(size int) {
    ccache.New(ccache.Configure[*typesystem.TypeSystem]().MaxSize(size))
}
```

?

Reply:
Seperation of concerns really. The TypesystemResolver doesn't care about the size of the cache or the kind of cache provided etc.. It just _needs a_ cache of a specific type. Since the server has the size available to it as server-level config, then it felt reasonable to have the server construct the cache it needs to provide the resolver with a cache.